src/lightning_app/core/api.py

```diff
@@ -59,16 +67,21 @@ class SessionMiddleware:
 
 
 class UIRefresher(Thread):
-    def __init__(self, api_publish_state_queue) -> None:
+    def __init__(self, api_publish_state_queue, api_commands_metadata_queue, api_commands_responses_queue) -> None:
         super().__init__(daemon=True)
         self.api_publish_state_queue = api_publish_state_queue
+        self.api_commands_metadata_queue = api_commands_metadata_queue
+        self.api_commands_responses_queue = api_commands_responses_queue
         self._exit_event = Event()
 
     def run(self):
         # TODO: Create multiple threads to handle the background logic
         # TODO: Investigate the use of `parallel=True`
-        while not self._exit_event.is_set():
-            self.run_once()
+        try:
+            while not self._exit_event.is_set():
+                self.run_once()
+        except Exception:
+            logger.error(traceback.print_exc())
```

Comment:
Shouldn’t we re-throw here? We’re having an unhandled exception go silent (except for logs) otherwise.

Reply:
Done !